### PR TITLE
feat(insights): capture an event when date exclusions change

### PR DESCRIPTION
**Pull request by sampennington**
## Problem

Nobody can tell how many people use the date picker's "Incomplete periods" toggle. The control shipped with no instrumentation, so the only two available signals both mislead:

- Autocapture clicks on `data-attr="date-filter-exclude-incomplete-periods"` measure clicks, not state — a user who toggles on then off looks identical to one who kept it on.
- Counting insights that store `dateRange.excludeIncompletePeriods` measures the *field*, not the control. That field is part of the public query schema, so the API, the MCP `insight-create` tool, and Max all write it. Insights carried it months before the toggle existed, which makes stored-field counts a wild overcount of UI adoption.

Answering "is this toggle discoverable?" currently takes a warehouse investigation, and the answer changes depending on which proxy you pick. The same gap applies to the day-of-week exclusion chips in the same flyout.

## Changes

- Adds `reportInsightDateExclusionsChanged` to `eventUsageLogic`, capturing `insight date exclusions changed` with `exclude_incomplete_periods`, `excluded_days_of_week_count`, and `query_kind`.
- Fires it from `InsightDateFilter` only when an exclusion actually changes, so re-opening the flyout or re-picking the same days emits nothing.
- Reports resulting state rather than a click, so on/off and keep/revert are distinguishable, and `query_kind` shows which insight types the control matters for.

Deliberately not changed: `data-attr` values stay frozen, per the `frontend/src/CLAUDE.md` rule that they are API. `LemonSwitch` also puts `data-attr` on its inner button while `label` renders as a sibling, so autocapture misses clicks on the label text — measured at 1 label click against 200 switch clicks, and this event makes that leak irrelevant for this control rather than worth a change to a shared design-system component.

No UI change, so no screenshots.

## How did you test this code?

No tests added. The change is one capture call on an existing kea action path; its siblings (`reportInsightDatePickerOpened`, `reportInsightDateRangeChanged`) have no tests either, and a test asserting "this action calls `posthog.capture`" would assert the implementation rather than a regression anyone could hit.

What I actually ran:

- `pnpm --filter=@posthog/frontend typescript:check` — no errors in either touched file. The run does report pre-existing errors elsewhere from unbuilt workspace packages (`@posthog/quill*`, `@posthog/hogvm`) in this environment; CI builds those.
- `pnpm --filter=@posthog/frontend typegen:file frontend/src/lib/utils/eventUsageLogic.ts` then `pnpm --filter=@posthog/frontend fix`, so the generated action type and formatting are in sync.

Not done: no manual browser verification, and `hogli ci:preflight` was unavailable in this environment.

Verifying the event after deploy: it should appear as `insight date exclusions changed`, with `exclude_incomplete_periods` splitting enables from disables.

## Automatic notifications

- [ ] Publish to changelog?

## Docs update

None — internal analytics event, no user-facing behavior or documented workflow changes.

## 🤖 Agent context

**Autonomy:** Human-driven (agent-assisted)

Written by the PostHog Slack app (Claude Code) from a Slack thread asking how many users had touched the incomplete-periods toggle.

The investigation is why this PR exists. Three successive attempts to answer that question from existing data each produced a different number, and each was wrong for a different reason: a project-scoped `system.insights` query undercounted by looking only at PostHog's own project; the cross-team fix then overcounted by attributing API- and agent-written `excludeIncompletePeriods` fields to the toggle; and the autocapture click count can't distinguish keeping the setting from reverting it. That's what convinced me the fix is an event rather than a better query.

I also checked and ruled out two theories along the way: the second exclusions control in the Quill date picker (`DateFilterExclusionsControl`) is not a live surface — its `quill-date-picker` flag has never returned `true` for any user — and the `LemonSwitch` label-click blind spot is real but immaterial here.

Skills invoked: none of the repo skills matched this change; I read `.github/pull_request_template.md`, `.agents/skills/writing-pr-descriptions/SKILL.md`, and `frontend/src/CLAUDE.md` directly for shape and conventions.

Nothing in this diff or description carries material from the originating session: the change touches only this repo's own code, and no customer names, identifiers, or figures from the analytics queries appear here.

---
*Created with [PostHog](https://posthog.com?ref=pr) from a [Slack thread](https://posthog.slack.com/archives/C0368RPHLQH/p1786955154540149?thread_ts=1786955154.540149&cid=C0368RPHLQH)*

### CODE REVIEW

File: frontend/src/scenes/insights/filters/InsightDateFilter/InsightDateFilter.tsx [hunk 1]
```diff
@@ -54,12 +54,17 @@ export function InsightDateFilter({ disabled }: InsightDateFilterProps): JSX.Ele
...
+        const incompleteChanged = next.incomplete !== exclusions.incomplete
+        if (daysChanged) {
             updateQuerySource(computeDaysOfWeekUpdate(nextExcludedDaysOfWeek, dateRange))
         }
-        if (next.incomplete !== exclusions.incomplete) {
+        if (incompleteChanged) {
             updateDateRange({ excludeIncompletePeriods: next.incomplete ? true : null }, true)
         }
+        if (daysChanged || incompleteChanged) {
+            reportInsightDateExclusionsChanged(querySource?.kind, next.incomplete, nextExcludedDaysOfWeek.length)
```
Comment (sampennington) [inline frontend/src/scenes/insights/filters/InsightDateFilter/InsightDateFilter.tsx]: 🤖 **sp-ship** · 💻📐 `[code+xp]` · **Blocker** · `mechanical`

`excluded_days_of_week_count` comes from the raw picker value, but `computeDaysOfWeekUpdate` normalizes a full-week exclusion back to `daysOfWeek: null` — exclude nothing. All seven chips toggle independently with no cap, so the seventh click makes `daysChanged` true, persists "no exclusion", resets the chips, and still reports `7`. That is a phantom bucket in the metric this event exists to measure.

Derive the count from what actually gets persisted:

```ts
const nextUpdate = daysChanged ? computeDaysOfWeekUpdate(nextExcludedDaysOfWeek, dateRange) : null
if (nextUpdate) {
    updateQuerySource(nextUpdate)
}
// ...
const persistedExcludedDays = nextUpdate ? getExcludedDaysOfWeek(nextUpdate.dateRange) : excludedDaysOfWeek
reportInsightDateExclusionsChanged(querySource?.kind, next.incomplete, persistedExcludedDays.length)
```

**Review by sampennington — COMMENTED** [review]
🤖 **sp-ship** · 🚫 Blockers

One blocker: the new `excluded_days_of_week_count` reports the raw picker selection, but `computeDaysOfWeekUpdate` normalizes a full-week exclusion to `daysOfWeek: null`, so the event can claim a state the insight never held. It is a one-expression fix inside the diff.

The rest is clean — the refactor is behaviour-preserving, the action and property names match their siblings in `eventUsageLogic.ts`, the existing `daysOfWeekFilterUtils` helpers are reused rather than reimplemented, and nothing new lands on the render path. The missing tests were considered and let go: every sibling `report*` action has zero coverage, and the only assertion available on the logic half is that a mock was called.

File: frontend/src/scenes/insights/filters/InsightDateFilter/InsightDateFilter.tsx (hunk 1 above)
Comment (sampennington) [reply frontend/src/scenes/insights/filters/InsightDateFilter/InsightDateFilter.tsx]: 🤖 **sp-triage**

Confirmed and fixed in `079e28b`. Verified the trigger: the seven chips in `LemonDateFilterExclusions.tsx` toggle independently with no cap, and `computeDaysOfWeekUpdate` maps `[1..7]` to `daysOfWeek: null` (pinned by `daysOfWeekFilterUtils.test.ts`), so the seventh click persisted "exclude nothing" while reporting `7`.

`handleExclusionsChange` now keeps the computed update and derives the count from it via `getExcludedDaysOfWeek(daysUpdate.dateRange)`, falling back to the current `excludedDaysOfWeek` when only the incomplete-periods toggle changed. Typecheck is clean.

[the remark this answers is not in the document]
Comment (sampennington) [reply frontend/src/scenes/insights/filters/InsightDateFilter/InsightDateFilter.tsx]: 🤖 **sp-triage**

Valid, and fixed in `079e28b`. `excluded_days_of_week_count` is now derived from the normalized update that gets persisted (`getExcludedDaysOfWeek(daysUpdate.dateRange)`) rather than the raw picker selection, so a full-week selection reports `0` to match the `daysOfWeek: null` that is actually stored.
